File: RotateArray.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    const n = nums.length;
    k = k % n;
    let ind, cur, temp;
    const cycle = gcd(k, n);
    for (let j = 0; j < cycle; j++) {
        ind = j;
        cur = nums[j];
        for (let i = 0; i < Math.floor(n / cycle); i++) {
            temp = nums[(ind + k) % n];
            nums[(ind + k) % n] = cur;
            ind = (ind + k) % n;
            cur = temp;
        }
    }
};

function gcd(a, b) {
    while (b !== 0) {
        const temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

File: RotateArray.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        
        int n=nums.size();
        k=k%n;
        int ind,cur,temp;
        int cycle=__gcd(k,n);
        for(int j=0;j<cycle;j++)
        {
            ind=j;
            cur=nums[j];

            for(int i=0;i<(n/cycle);i++)
            {
                temp=nums[(ind+k)%n];
                nums[(ind+k)%n]=cur;
                ind=(ind+k)%n;
                cur=temp;
                


                
            }
        }
    }
};
int main()
{
    
    return 0;
}